caas/kubernetes/provider/specs/legacy.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package specs
+
+import (
+	"strings"
+
+	"github.com/juju/errors"
+	"gopkg.in/yaml.v2"
+	core "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	k8syaml "k8s.io/apimachinery/pkg/util/yaml"
+
+	// "github.com/juju/juju/caas"
+	"github.com/juju/juju/caas/specs"
+)
+
+type podSpecLegacy struct {
+	CaaSSpec specs.PodSpecLegacy
+	K8sSpec  K8sPodSpecLegacy
+}
+
+// Validate is defined on ProviderPod.
+func (p podSpecLegacy) Validate() error {
+	if err := p.CaaSSpec.Validate(); err != nil {
+		return errors.Trace(err)
+	}
+	if err := p.K8sSpec.Validate(); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+func (p podSpecLegacy) ToLatest() *specs.PodSpec {
+	pSpec := &specs.PodSpec{}
+	pSpec.Version = specs.CurrentVersion
+	pSpec.OmitServiceFrontend = p.CaaSSpec.OmitServiceFrontend
```

Comment:
OmitServiceFrontend becomes part of the deployment service type defined in charm metadata.yaml

Reply:
left a TOD to remove `OmitServiceFrontend` from v3 when we deprecate the legacy version.